move-to-infra/move-to-infra.sh

```diff
@@ -0,0 +1,179 @@
+checkStatus()
+{
+  LABEL="$1"
+  NAMESPACE="$2"
+  EXPECTED_PODS="$3"
+  
+  echo "...checking pod labels ${LABEL} at namespace ${NAMESPACE}..."
+  ATTEMPTS=0
+  until [[ $ATTEMPTS -ge 10 ]]
+  do
+    PODS_READY=$(oc get pod  -l "${1}" --field-selector=status.phase==Running -o jsonpath="{.items[*].metadata.name}" -n${NAMESPACE} | wc -w)
+    if [[ $PODS_READY -eq $EXPECTED_PODS ]]; then
+      echo "...found expected pods running"
+      break
+    fi
+    ATTEMPTS=$[$ATTEMPTS+1]
+    echo "Sleep 10s and check again."
+    sleep 10
+  done
+  if [[ $PODS_READY -ne $EXPECTED_PODS ]]; then
+    echo "...not found expected pods. exit"
+    exit 1
+  fi
+}
+
+# create config map to move 
+monitoringConfigMap()
+{
+  cat <<EOT > cluster-monitoring-configmap.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |+
+    alertmanagerMain:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    prometheusK8s:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    prometheusOperator:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    grafana:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    k8sPrometheusAdapter:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    kubeStateMetrics:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    telemeterClient:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+    openshiftStateMetrics:
+      nodeSelector:
+        node-role.kubernetes.io/infra: ""
+      tolerations:
+      - key: infra
+        value: reserved
+        effect: NoSchedule
+      - key: infra
+        value: reserved
+        effect: NoExecute
+EOT
+}
+
+# exit in case of any errors
+set -e
```

Comment:
Consider using error traps to better manage any errors that might happen

Reply:
TRAP would be useful once any action to do before exiting script with error.
It is not the case here.